zebra-network/src/peer/handshake.rs

```diff
@@ -588,13 +588,25 @@ where
     debug!(?our_version, "sending initial version message");
     peer_conn.send(our_version).await?;
 
-    let remote_msg = peer_conn
+    let mut remote_msg = peer_conn
         .next()
         .await
         .ok_or(HandshakeError::ConnectionClosed)??;
-
-    // Check that we got a Version and destructure its fields into the local scope.
     debug!(?remote_msg, "got message from remote peer");
+
+    // Wait for next message if the one we have is not Version
+    match remote_msg {
+        Message::Version { .. } => (),
+        _ => {
+            remote_msg = peer_conn
+                .next()
+                .await
+                .ok_or(HandshakeError::ConnectionClosed)??;
+            debug!(?remote_msg, "got message from remote peer");
+        }
+    }
```

Comment:
Blocker: This is good, but it's not quite enough.

We need to ignore any number of extra non-Version messages, then continue with the handshake as soon as we get a Version message.

(Currently, this code only ignores 1 extra non-Version message.)